clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -561,14 +562,14 @@ scheduler_tmpl_t<RecordType, ReaderType>::set_initial_schedule(
             VPRINT(this, 2, "Assigning input #%d to output #%zd\n", i, index);
         }
     } else if (options_.mapping == MAP_TO_RECORDED_OUTPUT) {
-        if (outputs_.size() > 1) {
-            if (options_.replay_as_traced_istream == nullptr)
-                return STATUS_ERROR_INVALID_PARAMETER;
+        if (options_.replay_as_traced_istream != nullptr) {
             sched_type_t::scheduler_status_t status = read_traced_schedule();
```

Comment:
So now control may flow here for `output_.size() == 1` too? There's no schedule as-such for a single output stream right?

Reply:
I found it useful to replay the schedule for one output for some manual tests and there's no real reason to not allow it if the user wants to do it: but yes it should match the analyzer serial mode so there's no big benefit to reading the schedule file.

Reply:
Added a comment.